Automattic/wp-calypso
# Happychat: Refactor to use middleware (attempt 2)

**This is now replaced with #12218. See Automattic/happychat#275**

Depends on #11821 

This PR has minimal functional changes and is mostly a refactor.

If a chat was active in the last 10 minutes and an operator is still available, the Happychat icon will now be shown in the bottom left of the sidebar after boot. We can bump the timeout higher once we're confident that the service can handle a higher number of connections.

This was originally #11381 but had to be reverted in #11561. Commit e38a250 fixes the issue that made the revert necessary.

### Testing instructions

1. Run automated tests with `npm run test-client client/state/happychat`
1. Confirm that chat works as usual
1. Reload Calypso, confirm that Calypso automatically reconnects to Happychat (the transcript is not yet shown).
1. Wait at least 10 minutes after sending a message, then refresh Calypso again. Confirm that Calypso does not automatically connect to Happychat, until you visit /help/contact.
